controlplane/kubeadm/internal/workload_cluster_test.go

```diff
@@ -1260,6 +1262,69 @@ func TestClusterStatus(t *testing.T) {
 	}
 }
 
+func TestUpdateFeatureGatesInKubeadmConfigMap(t *testing.T) {
+	tests := []struct {
+		name                     string
+		clusterConfigurationData string
+		newFeatureGates          map[string]bool
+		wantFeatureGates         map[string]bool
+	}{
+		{
+			name: "it updates feature gates",
+			clusterConfigurationData: yaml.Raw(`
+				apiVersion: kubeadm.k8s.io/v1beta2
+				kind: ClusterConfiguration`),
+			newFeatureGates:  map[string]bool{"EtcdLearnerMode": true},
+			wantFeatureGates: map[string]bool{"EtcdLearnerMode": true},
+		},
+		{
+			name: "it should preserve existing feature gates if new value is nil",
```

Comment:
not sure about this, shouldn't we remove feature gates if a user removes them all?

Reply:
done in combination with the previous comment